### PR TITLE
Add an architectures arg to go_toolchain()

### PR DESCRIPTION
**Pull request by Tatskaari**
Fixes:  #1579

### CODE REVIEW

File: rules/go_rules.build_defs [hunk 1]
```diff
@@ -20,6 +21,9 @@ def go_toolchain(name:str, url:str|dict = '', version:str = '', hashes:list = []
                      but not both.
       hashes (list): A list of possible hashes for the downloaded archive. Optional.
       visibility (list): Visibility specification. Defaults to public.
+      architectures (dict): Any additional architectures to install in go architecture format e.g. linux_amd64. This
```
Comment (peterebden) [inline rules/go_rules.build_defs:24]: says `dict` but should be `list`?

Comment (Tatskaari) [reply rules/go_rules.build_defs:24]: Ah good catch

**Comment by sagikazarmark** [comment]
Is this a potential fix for #1579 ? Or an additional feature?

**Comment by Tatskaari** [comment]
> Is this a potential fix for #1579 ? Or an additional feature?

Oh yeah I think so. If you're doing `///{os}_{arch}//some_target:test` and not doing `--arch {os}_{arch}`, then go_toolchain() won't install the SDK for that arch, and it will fail as you've described in your issue.

**Pull request merged**